pkg/web/api.go

```diff
@@ -0,0 +1,45 @@
+package web
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+type MigrateParams struct {
+	SourceHost       string            `json:"source_host,omitempty"`
+	TargetHost       string            `json:"target_target,omitempty"`
+	ContainerName    string            `json:"container_name,omitempty"`
+	SourceUser       string            `json:"source_user,omitempty"`
+	TargetUser       string            `json:"target_user,omitempty"`
+	ExcludePaths     []string          `json:"excluded_paths,omitempty"`
+	TcpPorts         map[uint16]uint16 `json:"tcp_ports,omitempty"`
+	ExcludedTcpPorts []uint16          `json:"excluded_tcp_ports,omitempty"`
+	ForceCreate      bool              `json:"force_create,omitempty"`
+	DisableStart     bool              `json:"disable_start,omitempty"`
+	Debug            bool              `json:"debug,omitempty"`
+}
+
+func MigrateMachine(request *http.Request) (interface{}, error) {
+	var (
+		migrateParams MigrateParams
+		data          interface{}
+	)
+
+	if err := json.NewDecoder(request.Body).Decode(&migrateParams); err != nil {
+		return nil, err
+	}
+
+	// place for executor
+	// Executor.execute // or something
+	output_mock := []byte("{\"foo\": \"bar\"}")
+	// return result to client
+
+	if json.Unmarshal(output_mock, &data) != nil {
```

Comment:
The golang pattern I've seen out there is `if err := X(); err != nil {...}`. It seems to me that we could use this pattern here as well. As a result, we'd have a more specific error message ("Invalid json output from executor" does the trick, but it might be too generic).

Reply:
hmm in that case I wanted to have this generic "Invalid json output from executor", because in my opinion it's better than return "invalid character 'd' after object key:value pair" which is original err message. The reason why is I have no idea if I received this message because of bad request or bad response.

However I can agree with you and right now let's send original err message

Reply:
done

Reply:
Good point. As far as I know, the recommended way is to just wrap the error you got from `Unmarshal()` into your own error. Here is an example of how to do that: